lib/rspec/core/example_group.rb

```diff
@@ -650,6 +650,16 @@ def self.method_missing(name, *args)
       end
       private_class_method :method_missing
 
+      # @private
+      def self.method_added(method_name)
+        if (@added_methods ||= Set.new).include?(method_name)
```

Comment:
Given that users can write class methods (e.g. macros) in their example groups, and such methods would run with access to the `@added_methods`, I'd prefer to see this use `@__added_methods` to reduce the likelihood of naming conventions.